airflow/smart_sensor_dags/smart_sensor_group.py

```diff
@@ -0,0 +1,63 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+""" Smart sensor DAGs managing all smart sensor tasks """
+
+from builtins import range
+from datetime import timedelta
+
+from airflow.configuration import conf
+from airflow.models import DAG
+from airflow.sensors.smart_sensor_operator import SmartSensorOperator
+from airflow.utils.dates import days_ago
+
+args = {
+    'owner': 'airflow',
+    'start_date': days_ago(2),
+}
+
+num_smart_sensor_shard = conf.getint("smart_sensor", "shards")
+shard_code_upper_limit = conf.getint('smart_sensor', 'shard_code_upper_limit')
+
+for i in range(num_smart_sensor_shard):
+    shard_min = (i * shard_code_upper_limit) / num_smart_sensor_shard
+    shard_max = ((i + 1) * shard_code_upper_limit) / num_smart_sensor_shard
+
+    dag_id = 'smart_sensor_group_shard_{}'.format(i)
+    dag = DAG(
+        dag_id=dag_id,
+        default_args=args,
+        schedule_interval=timedelta(days=1),
```

Comment:
```suggestion
        schedule_interval=timedelta(days=1),
        start_date=days_ago(2),
```

Reply:
The same start_date was defined in default_args.

Reply:
`start_date` should only be used on task level if a user likes to have a different `start_date` for that particular task. Adding `start_date` in `default_args` basically means all tasks have same start_date, so it is better to define them at DAG level instead. Check https://github.com/apache/airflow/pull/9982 for more details